prov/rxm/src/rxm.h

```diff
@@ -420,12 +420,30 @@ struct rxm_buf_pool {
 	fastlock_t lock;
 };
 
+struct rxm_msg_eq_entry {
+	struct slist_entry	slist_entry;
+	ssize_t			rd;
+	uint32_t		event;
+	/* Used for connection refusal */
+	void			*context;
+	/* must stay at the bottom */
+	struct fi_eq_cm_entry	cm_entry;
+};
+
+#define RXM_MSG_EQ_ENTRY_SZ (sizeof(struct rxm_msg_eq_entry) + \
+			     sizeof(struct rxm_cm_data))
+#define RXM_CM_ENTRY_SZ (sizeof(struct fi_eq_cm_entry) + \
+			 sizeof(struct rxm_cm_data))
+
 struct rxm_ep {
 	struct util_ep 		util_ep;
 	struct fi_info 		*rxm_info;
 	struct fi_info 		*msg_info;
 	struct fid_pep 		*msg_pep;
 	struct fid_eq 		*msg_eq;
+	struct slist		msg_eq_entry_list;
+	fastlock_t		msg_eq_entry_list_lock;
+	struct util_buf_pool	*msg_eq_entry_pool;
```

Comment:
Do we need a pool for these, or can we just alloc/free as needed, to avoid consuming extra memory?

Reply:
Yes, we can. Is it okay if we have free on fast path? The alloc would not be in fast path.